lib/promscrape/discovery/docker/container.go

```diff
@@ -67,14 +69,51 @@ func parseContainers(data []byte) ([]container, error) {
 	return containers, nil
 }
 
-func addContainersLabels(containers []container, networkLabels map[string]*promutils.Labels, defaultPort int, hostNetworkingHost string) []*promutils.Labels {
+func addContainersLabels(containers []container, networkLabels map[string]*promutils.Labels, defaultPort int, hostNetworkingHost string, matchFirstNetwork bool) []*promutils.Labels {
+	allContainers := make(map[string]container)
```

Comment:
I'd suggest to rename it for `containersByID`

Reply:
And use `map[string]*container` for storing results:

```go
var containersByID map[string]*container
if matchFirstNetwork{
  containersByID = make(map[string]*container)
  for _, c := range containers {
		containersByID[c.ID] = c
  }
}
```

Reply:
But one container could contains multiple ports. For example, if we have networks and container like this:
```json
[{
		"name": "network_1",
		"id": "...",
		"ip": "192.168.0.1"
	},
	{
		"name": "network_2 ",
		"id": "...",
		"ip": "192.168.0.2"
	}
]
```

```[{
		"name": "container_1",
		"ports": [80, 8080],
		"network": ["network_1", "network_2"]
	},
	{
		"name": "container_2",
		"ports": [90, 9090],
		"network": ["network_1", "network_2"]
	}
]
```

We will have 8 scrape targets:
```
192.168.0.1:80  // c1 + p1 + n1
192.168.0.1:8080  // c1 + p2 + n1
192.168.0.2:80  // c1 + p1 + n2
192.168.0.2:8080  // c1 + p2 + n2
192.168.0.1:90  // ...
192.168.0.1:9090 // ...
192.168.0.2:90 // ...
192.168.0.2:9090 // ...
```